spec/models/user_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  subject(:user) { FactoryBot.build(:user)}
+
+  it 'has a valid factory' do
+    expect(user.validate!).to eq(true)
+  end
+
+  it { is_expected.to have_many(:completed_lessons) }
+
+  it { is_expected.to have_many(:lessons).through(:completed_lessons) }
+
+  it { is_expected.to validate_presence_of(:email) }
+
+  it 'invalidates email format with no @' do
+    user.email = 'email.com'
+    expect(user.valid?).to eq(false)
+  end
+
+  it 'invalidates email format with no .' do
+    user.email = 'user@email'
+    expect(user.valid?).to eq(false)
+  end
+
+  it 'validates email format' do
+    user.email = 'user@email.com'
+    expect(user.valid?).to eq(true)
+  end
```

Comment:
This test is not needed because the `has a valid factory` test already covers it